pkg/azure/api/providerspec.go

```diff
@@ -54,6 +54,10 @@ const (
 	// MachineSetKindVMO is the machine set kind for VirtualMachineScaleSet Orchestration Mode VM (VMO).
 	// Deprecated: Use AzureVirtualMachineProperties.VirtualMachineScaleSet instead.
 	MachineSetKindVMO string = "vmo"
+
+	// SkipMarketplaceAgreementAnnotation is an annotation for the machineclass. When this is found on a MC, the controller will skip
+	// checking for the marketplace license agreement.
+	SkipMarketplaceAgreementAnnotation = "beta.azure.machine.sapcloud.io/skip-marketplace-agreement"
```

Comment:
why do you need this annotation? We can do without it as well.

Reply:
Annotation let me have a feature that is not committed to the spec, which I think is a good thing given that the spec is not really versioned. 

The main reasoning is this: In an ideal world you would not need this annotation. We want to skip the license agreement check because we can't find a way to reliably list the image using the go client (but we can use it in a VM). Given alternatives from the image spec (if we could list it) I would use them instead of this annotation. Hence, this should be replaced when there is some better support from the SDK, or until we figure better way to do it. Cycling back to the first argument, I wouldn't want to commit this to spec.